tensorrt_llm/_torch/pyexecutor/decoder.py

```diff
@@ -240,153 +225,162 @@ def _handle_stop_criteria(self, request: LlmRequest, new_token: int,
 
         return False
 
-    def update_requests(self, scheduled_requests: ScheduledRequests,
-                        new_tensors_host: Dict[str, torch.tensor],
-                        decoder_event: torch.cuda.Event):
-        if decoder_event:
-            decoder_event.synchronize()
-        new_tokens_list = new_tensors_host["new_tokens_host"].tolist()
+    def update_requests(self, decoder_state: DecoderState) -> None:
+        if decoder_state.decoder_event:
+            decoder_state.decoder_event.synchronize()
+        new_tokens_list = decoder_state.new_tensors_host[
+            "new_tokens_host"].tolist()
 
-        idx = 0
+        request_idx = 0
+        logits_idx = 0
```

Comment:
How about renaming `logits_idx` to `token_idx`? It's used to find a token in `new_tokens_list`.

Reply:
I reverted it back to `idx` in this PR. The rename is only reasonable with the addition of logits handling code.